doc/user/content/guides/cdc-postgres.md

```diff
@@ -0,0 +1,260 @@
+
+### Create a source
+
+Postgres sources are materialized by default and ingest the raw replication stream data for all tables included in the publication, to avoid creating multiple replication slots and minimize the required bandwidth. To create a source:
+
+```sql
+CREATE MATERIALIZED SOURCE mz_source
+FROM POSTGRES
+  CONNECTION 'host=postgres port=5432 user=host dbname=postgres sslmode=require'
+PUBLICATION 'mz_source';
+```
+
+{{< note >}}
+Materialize performs an initial sync of the tables included in the publication before it starts ingesting change events, so you should make sure that your source tables **fit into memory**.
+{{</ note >}}
+
+In practice, the `mz_source` looks like:
+
+```sql
+SHOW COLUMNS FROM mz_source;
+
+   name   | nullable |  type
+----------+----------+---------
+ oid      | f        | integer
+ row_data | f        | list
+```
+
+Each row of every upstream table is represented as a single row with two columns in the replication stream source:
+
+* `oid`: A unique identifier for the tables included in the publication;
+
+* `row_data`: A text-encoded, variable length `list`. The number of text elements in a list is always equal to the number of columns in the upstream table.
+
+The next step is to break down this source into views that reproduce the publication’s original tables and can be used as a base for your materialized views.
+
+#### Create replication views
+
+Once you've created the Postgres source, you can create views that filter the replication stream based on the `oid` identifier and convert the text elements in `row_data` to the original data types:
+
+_Create views for specific tables included in the Postgres publication:_
+
+```sql
+CREATE VIEWS FROM SOURCE mz_source (view1, view2);
+```
+
+_Create views for all tables:_
+
+```sql
+CREATE VIEWS FROM SOURCE mz_source;
+```
+
+Under the hood, Materilize parses this statement into view definitions for each table (so you don't have to!).
+
+### Create a materialized view
+
+Any materialized view defined on top of this source will be incrementally updated as new change events stream in, as a result of `INSERT`, `UPDATE` and `DELETE` operations in the original Postgres database.
+
+```sql
+CREATE MATERIALIZED VIEW cnt_view1 AS
+SELECT field1,
+       COUNT(*) AS cnt
+FROM view1
+GROUP BY field1;
+```
+
+## Kafka + Debezium
+
+If Kafka is part of your stack, you can use [Debezium](https://debezium.io/) and the [Kafka source](/sql/create-source/avro-kafka/#debezium-envelope-details) to propagate CDC data from Postgres to Materialize. Debezium captures row-level changes resulting from `INSERT`, `UPDATE` and `DELETE` operations in the upstream database and publishes them as events to Kafka using Kafka Connect-compatible connectors.
+
+### Setup
+
+**Minimum requirements:** Postgres 10+
```

Comment:
This is not _technically_ true, since Debezium also supports Postgres 9.6, but this requires users to install decoding plugins (while from 10 we can just directly use `pgoutput`) and this version will reach EOL soon (November 2021).